internal/store/postgres/approval_repository.go

```diff
@@ -37,19 +37,32 @@ func (r *ApprovalRepository) ListApprovals(ctx context.Context, conditions *doma
 	records := []*domain.Approval{}
 
 	db := r.db.WithContext(ctx)
-	db = db.Preload("Appeal.Resource")
-	db = db.Joins("Appeal")
-	db = db.Joins(`JOIN "approvers" ON "approvals"."id" = "approvers"."approval_id"`)
+	db = db.Joins("Appeal").
+		Joins("Appeal.Resource").
+		Joins(`JOIN "approvers" ON "approvals"."id" = "approvers"."approval_id"`)
 
 	if conditions.CreatedBy != "" {
 		db = db.Where(`"approvers"."email" = ?`, conditions.CreatedBy)
 	}
 	if conditions.Statuses != nil {
 		db = db.Where(`"approvals"."status" IN ?`, conditions.Statuses)
 	}
+	if conditions.Q != "" {
+		db = db.Where(db.
+			Or(`"Appeal"."account_id" LIKE ?`, fmt.Sprintf("%%%s%%", conditions.Q)).
+			Or(`"Appeal"."role" LIKE ?`, fmt.Sprintf("%%%s%%", conditions.Q)).
+			Or(`"Appeal__Resource"."urn" LIKE ?`, fmt.Sprintf("%%%s%%", conditions.Q)),
```

Comment:
Is `Q` being used for generic string input for search kind of feature?  
basically search string in any of these three fields... ?

Reply:
yes, please check https://github.com/goto/guardian/issues/52